entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```
@@ -289,20 +289,127 @@ private void doUpdate(RequestContext requestContext, EntityUpdateRequest request
+    for (String entityId : entitiesMap.keySet()) {
+      Map<String, Document> transformedUpdateOperations =
+          transformUpdateOperations(
+              entitiesMap.get(entityId).getUpdateOperationList(), requestContext);
+      if (transformedUpdateOperations.isEmpty()) {
+        continue;
+      }
+      entitiesUpdateMap.put(
+          new SingleValueKey(requestContext.getTenantId().orElseThrow(), entityId),
+          transformedUpdateOperations);
+    }
+
+    if (entitiesUpdateMap.isEmpty()) {
+      LOG.error("There are no entities to update!");
+      return;
+    }
+
+    try {
+      entitiesCollection.bulkUpdateSubDocs(entitiesUpdateMap);
+    } catch (Exception e) {
+      LOG.error("Failed to update entities {}", entitiesMap, e);
+      throw e;
+    }
+  }
+
+  private Map<String, Document> transformUpdateOperations(
+      List<UpdateOperation> updateOperationList, RequestContext requestContext) {
+    Map<String, Document> documentMap = new HashMap<>();
+    for (UpdateOperation updateOperation : updateOperationList) {
+      if (!updateOperation.hasSetAttribute()) {
+        continue;
+      }
+      SetAttribute setAttribute = updateOperation.getSetAttribute();
+      String attributeId = setAttribute.getAttribute().getColumnName();
+      String subDocPath =
+          entityAttributeMapping
+              .getDocStorePathByAttributeId(requestContext, attributeId)
+              .orElseThrow(
+                  () -> new IllegalArgumentException("Unknown attribute FQN " + attributeId));
+      AttributeValue attributeValue =
+          EntityQueryConverter.convertToAttributeValue(setAttribute.getValue()).build();
+      try {
+        String jsonValue = DocStoreJsonFormat.printer().print(attributeValue);
+        documentMap.put(subDocPath, new JSONDocument(jsonValue));
+      } catch (Exception e) {
+        LOG.warn("Failed to put update corresponding to {} in the documentMap", subDocPath, e);
```

We should throw the exception back here

---

```suggestion
        LOG.error("Failed to put update corresponding to {} in the documentMap", subDocPath, e);
```